packages/plugin-ext/src/hosted/browser/main-plugin-service.ts

```diff
@@ -0,0 +1,45 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { interfaces } from 'inversify';
+import { KeysToKeysToAnyValue } from '../../common/types';
+import { FileStat } from '@theia/filesystem/lib/common/filesystem';
+import { ExtPluginApi } from '../../common/plugin-ext-api-contribution';
+import { PluginMetadata } from '../../common/plugin-protocol';
+
+export type DebugActivationEvent = 'onDebugResolve' | 'onDebugInitialConfigurations' | 'onDebugAdapterProtocolTracker';
+
+export const MainPluginService = Symbol('MainPluginService');
+export interface MainPluginService {
```

Comment:
Is there a way to avoid boilerplate interfaces and symbols or it is somehow require to break the cycle?

btw i like `MainPluginService`, it would be good to rename `HostedPluginSupport` to it and change a file name as well.

Reply:
For me no, at least I cannot think up a better solution.

For name, yes I plan to rename everything in `plugin-ext` with this plan https://github.com/theia-ide/theia/issues/3871#issuecomment-489557854